src/components/RightPane/SectionTable/SectionTableButtons.tsx

```diff
@@ -85,7 +85,7 @@ export const ScheduleAddCell = withStyles(styles)((props: ScheduleAddCellProps)
                     action: analyticsEnum.classSearch.actions.ADD_SPECIFIC,
                 });
             }
-            section.color = addCourse(section, courseDetails, term, scheduleIndex);
+            section.color = addCourse(section, courseDetails, term, scheduleIndex).section.color;
```

Comment:
```suggestion
            const newCourse = addCourse(section, courseDetails, term, scheduleIndex).section.color;
            section.color = newCourse.section.color
```
Optional: you can a temp variable to make it clearer